### PR TITLE
ci: add semantic release workflow

### PR DESCRIPTION
**Pull request by jim80net**
### Pre requirements:
- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/lightdash/lightdash/blob/main/.github/CONTRIBUTING.md#sending-a-pull-request).

### Reference:
Closes: #1286

### Description:
Merge to `main` to publish a release. 
Requires using[Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) in order to categorize the appropriate release type:
```
fix: a commit of the type fix patches a bug in your codebase (this correlates with [PATCH](http://semver.org/#summary) in Semantic Versioning).
feat: a commit of the type feat introduces a new feature to the codebase (this correlates with [MINOR](http://semver.org/#summary) in Semantic Versioning).
BREAKING CHANGE: a commit that has a footer BREAKING CHANGE:, or appends a ! after the type/scope, introduces a breaking API change (correlating with [MAJOR](http://semver.org/#summary) in Semantic Versioning). A BREAKING CHANGE can be part of commits of any type.
```

Publishing a release will:
- update the package.jsons
- write out the commits (categorized by features, fixes, and breaking changes) into the CHANGELOG.md
- publish to github
- post release updates to PRs and issues

**NOTE: Merging this PR may produce a release depending on what is in main between the last release and HEAD.** 

### Scope of the changes (select all that apply):
- [ ] Frontend  
- [ ] Backend  
- [ ] Common  
- [ ] E2E tests  
- [ ] Docs  
- [x] CI/CD

### CODE REVIEW

**Comment by ZeRego** [comment]
@owlas I noticed our PR check for semantic pull requests doesn't enforce the title to be correct. It might checking title or commits. In their [docs](https://github.com/zeke/semantic-pull-requests#configuration) they mention how we can change the configuration to only check the title. We could also make this check required. 

@owlas @jim80net wondering if we should/could skip the release if the PR is for `docs:` 🤔

**Comment by owlas** [comment]
The pull request bot we use: https://github.com/zeke/semantic-pull-requests

Checks that the PR title OR _at least one_ commit has a semantic commit format. @jim80net is that enough for the pipeline we have?

Definitely open to introduce a config file to scope the available semantic tags (e.g. only fix, feat, ...).

Alternatively we could do this in GitHub actions instead of a bot: https://github.com/amannn/action-semantic-pull-request

**Comment by ZeRego** [comment]
> Checks that the PR title OR _at least one_ commit has a semantic commit format. @jim80net is that enough for the pipeline we have?

@owlas this is not enough because we squash the PR. So there will only be 1 commit which this flow picks up. We need to force the title to be semantically correct. 

> @ZeRego are you happy that we'll simply not use the BREAKING CHANGE tag? The problem here is we do release breaking changes but we don't want to bump to 1.0.0

Yes, we just continue with what we are doing.

**Comment by owlas** [comment]
Just testing with @ZeRego - the annoying edge cases with the bot are:

1. single commit not conventional - PR title *is* conventional - fails (correctly because would add a non-conventional commit to `main` but not user friendly)
2. many commits with only one conventional - PR title *is not* conventional - passes but adds a non-conventional commit to `main`

This was a tricky edge case for the PR name bot too:
https://github.com/amannn/action-semantic-pull-request/pull/158

File: release.config.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    [
+      '@semantic-release/changelog',
+      {
+        'changelogFile': 'CHANGELOG.md',
+      }
+    ],
+    [
+      '@amanda-mitchell/semantic-release-npm-multiple',
+      {
+        'registries': {
```
Comment (owlas) [inline release.config.js:15]: This is great @jim80net ! And also gives us a short path to automating npm publishing when we need it 👍

**Review by owlas — APPROVED** [review]

**Comment by jim80net** [comment]
> Just testing with @ZeRego - the annoying edge cases with the bot are:
> 
> 1. single commit not conventional - PR title _is_ conventional - fails (correctly because would add a non-conventional commit to `main` but not user friendly since it needs a `git commit --amend`)
> 2. many commits with only one conventional - PR title _is not_ conventional - passes but adds a non-conventional commit to `main` (@ZeRego mentioned we can configure the bot to not allow this)
> 
> This was a tricky edge case for the PR name bot too: [amannn/action-semantic-pull-request#158](https://github.com/amannn/action-semantic-pull-request/pull/158)
> 
> @jim80net any ideas on the smoothest approach for this?

I think helping maintainers avoid mistakes is definitely a good call. I've submitted a PR that uses the action-semantic-pull-request. https://github.com/lightdash/lightdash/pull/1306

Regarding smooth approach, there is one type of annoyance with this system: the commit that didn't trigger a release that should have. If the check fails to catch a situation where the merge fails to produce a release, then generally what I recommend is to either: 
- revert the commit and push the commit again with an amended message
- push an empty commit with the correct type and an apologetic note (`git commit -m'fix: Wobbles now parse correctly (empty release trigger for a9381fed)`

While it doesn't make for the prettiest history, I think in the long run, the better workflow makes up for it.

**Comment by ZeRego** [comment]
@jim80net is there a way to skip the release ? This is useful when we are committing changes to docs or chores.

 E.g  `chore: update readme instructions [skip ci]`

**Comment by jim80net** [comment]
> @jim80net is there a way to skip the release ? This is useful when we are committing changes to docs or chores.
> 
> E.g `chore: update readme instructions [skip ci]`
> 
> -- Update --
> 
> Looks like github does this for us 🤔 https://github.blog/changelog/2021-02-08-github-actions-skip-pull-request-and-push-workflows-with-skip-ci/

So, semantic-release will only publish a release when there are commits between the last release and HEAD that are either `fix`, `feat`, include `BREAKING CHANGE` or have the `!` character at then end of the type (for example `refactor!: drop support for Node 6`).. 

If you would like to tell github actions not to run the workflows, you can use `[skip ci]` as you noted above.

**Pull request merged**